include/revng/Yield/CallGraphs/Graph.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/Support/BasicBlockID.h"
+#include "revng/Yield/Support/GraphLayout/Graphs.h"
+
+namespace yield::calls {
+
+struct Node {
+  BasicBlockID Address;
+  BasicBlockID NextAddress;
+
+  explicit Node(BasicBlockID Address = BasicBlockID::invalid()) :
+    Address(std::move(Address)) {}
+};
+
+struct Edge {
+  bool IsBackwards = false;
+};
+
+using PreLayoutGraph = layout::InputGraph<Node, Edge>;
+using PreLayoutNode = PreLayoutGraph::Node;
+
+using PostLayoutGraph = layout::OutputGraph<Node, Edge>;
+using PostLayoutNode = PostLayoutGraph::Node;
```

Comment:
Couldn't all these be the output version and then we do the layouting in place? AFAICT the layout engine work on its own internal graph, so this should be feasible.

Reply:
Wait, are these three the same? Is something weird going on with github today?

(closing this one too)